nodebuilder/tests/fraud_test.go

```diff
@@ -98,22 +100,43 @@ func TestFraudProofHandling(t *testing.T) {
 	select {
 	case p := <-subscr:
 		require.Equal(t, 10, int(p.Height()))
+		t.Log("Caught the proof....")
 		subCancel()
 	case <-ctx.Done():
 		subCancel()
 		t.Fatal("full node did not receive a fraud proof in time")
 	}
 
+	getCtx, getCancel := context.WithTimeout(ctx, time.Second)
+	proofs, err := fullClient.Fraud.Get(getCtx, byzantine.BadEncoding)
+	getCancel()
+
+	require.NoError(t, err)
+	require.NotNil(t, proofs)
+	require.Len(t, proofs, 1)
+	require.True(t, proofs[0].Type() == byzantine.BadEncoding)
 	// This is an obscure way to check if the Syncer was stopped.
 	// If we cannot get a height header within a timeframe it means the syncer was stopped
 	// FIXME: Eventually, this should be a check on service registry managing and keeping
 	//  lifecycles of each Module.
 	// 6.
-	syncCtx, syncCancel := context.WithTimeout(context.Background(), blockTime*5)
-	_, err = fullClient.Header.WaitForHeight(syncCtx, 15)
-	require.ErrorIs(t, err, context.DeadlineExceeded)
-	syncCancel()
+	attempts := 0
+	timeOut := blockTime * 5
+	height := uint64(15)
+	for attempts < 5 {
+		syncCtx, syncCancel := context.WithTimeout(context.Background(), timeOut)
+		_, err = full.HeaderServ.WaitForHeight(syncCtx, height)
+		syncCancel()
+		if err != nil {
+			break
+		}
+		attempts++
+		timeOut = blockTime
+		height++
+	}
```

Comment:
It is kinda confusing, why loop limit is based on `attempts`, while `height` is also iterated.  Whould it be same with `height < 20` condition?

Reply:
because during befp verification you can still receive several headers and I guess it depends on your hardware. Once, befp is validated, then syncer will be stopped